pkg/volume/rbd/attacher.go

```diff
@@ -214,12 +215,25 @@ func (detacher *rbdDetacher) UnmountDevice(deviceMountPath string) error {
 	}
 	klog.V(3).Infof("rbd: successfully umount device mountpath %s", deviceMountPath)
 
+	if devicePath == "" {
+		deviceMountedPathSeps := strings.Split(deviceMountPath, "-image-")
```

Comment:
It is possible that if global mount path is unmounted, but kubelet fails to unmap rbd device, then in future retries, it couldn't know rbd device by searching mount points because global mount path is unmounted.

This PR falls back to search by image pool and name. IMO it is a temporary fix until we can fix in higher level (e.g. separating unmount and detach). What do you think? cc @jsafrane @rootfs

Reply:
you can't separate umount and detach, they both must run in kubelet as part of `UnmountDevice`.

The detach we have is for 3rd party detach that we don't really support for RBD.